nvflare/private/fed/server/server_engine.py

```diff
@@ -350,8 +350,6 @@ def remove_clients(self, clients: List[str]) -> str:
     def _remove_dead_client(self, token):
         client = self.server.client_manager.remove_client(token)
         self.server.remove_client_data(token)
-        if self.server.admin_server:
```

Comment:
admin_server need to remove the dead client as well. We need this step.

Reply:
The root cause of this shutdown issue is the wrong order in the _shutdown_app_on_clients() in the server/training_cmds.py.  The "shutdown" message should be sent to the clients first, then call the engine.remove_clients(). Like below:

        replies = self.send_request_to_clients(conn, message)
        self.process_replies_to_table(conn, replies)

        err = engine.remove_clients(clients)
        if err:
            conn.append_error(err)
            return False